src/index.js

```diff
@@ -48,6 +49,14 @@ class App {
     this.elements.clearButton.addEventListener('click', () => {
       this.handleClearSearch();
     })
+
+    this.elements.favoritesSelect.addEventListener('change', (e) => {
+      const select = e.target;
+      const option = select.options[select.selectedIndex];
+      const location = JSON.parse(option.dataset.location);
+
+      if (select.value !== 'favorites') this.buildPageByQuery(location);
+    })
```

Comment:
```js
// index.js
 this.elements.favoritesSelect.addEventListener('change', (e) => {
      const city = e.target.value;
      
      if (city !== 'favorites') {
        const location = this.favorites.getCityLocation(city);
        
        if (location) {
          this.buildPageByQuery(location);
        }
      }
})

// Favorites.js

// Returns parsed array of favorites from localStorage or empty array in case there are no favorites yet
getFavorites() {
  const favorites = localStorage.getItem('favorites')
  return favorites ? JSON.parse(favorites) : []
}

// Returns city object or undefined in case it cannot be found
getCityLocation(cityName) {
  const favorites = this.getFavorites()
  return favorites.find(city => city.name === cityName)
}

addToFavorites(name, lat, lon) {
  const favorites = this.getFavorites()
  localStorage.setItem('favorites', JSON.stringify([
    ...favorites,
    { name, lat, lon }
  ]))
}

removeFromFavorites(name) {
  // Try it yourself
}

isFavorite(city) {
  return this.getCityLocation(city) !== undefined
}
```